lib/propshaft/asset.rb

```diff
@@ -2,10 +2,16 @@
 require "action_dispatch/http/mime_type"
 
 class Propshaft::Asset
+  PREDIGESTED_REGEX = /-([0-9a-zA-Z]{7,128})\.digested/
+
   attr_reader :path, :logical_path, :version
 
   def initialize(path, logical_path:, version: nil)
-    @path, @logical_path, @version = path, Pathname.new(logical_path), version
+    @path = path
+    @digest = logical_path.to_s[PREDIGESTED_REGEX, 1]
+    @digested_path = Pathname.new(logical_path) if @digest
+    @logical_path = Pathname.new(@digest ? logical_path.sub("-#{@digest}", "") : logical_path)
+    @version = version
```

Comment:
Now that we are normalizing the logical path to never have a digest in it, I believe that we can entirely discard the `.digested` suffix? It's already done its job in telling Propshaft that there is a digest in the file name. This would make non-digested and pre-digested logical paths look the same, and I think it would also incentivize the rest of the code to treat them the same.

By the same token, I think we don't need the `@digested_path` instance variable. Removing it (and having the logical path always `sub` instead of using a ternary operator) means extra `subs` that could have been avoided like you did here, but it also makes the code a bit tidier. At this point where we are still experimenting, I think that would be worth more than the small optimization and make it easier for others since the behavior is always the same.

```ruby
@logical_path =  logical_path.sub("-#{@digest}.digested", "")
```